core/include/detray/core/detector.hpp

```diff
@@ -130,23 +136,23 @@ class detector {
     using volume_finder =
         typename metadata::template volume_finder<container_t>;
 
-    using detector_view_type =
-        detector_view<metadata, covfie::field, host_container_types>;
+    using this_t = detector<metadata, bfield_t, container_t>;
+    using detector_view_type = detector_view<bfield_t, this_t>;
 
     using free_vector_type =
         typename free_track_parameters<transform3>::vector_type;
     using bound_vector_type =
         typename bound_track_parameters<transform3>::vector_type;
 
     /// Detector view types
-    using view_type = dmulti_view<dvector_view<volume_type>,
-                                  typename transform_container::view_type,
-                                  typename mask_container::view_type,
-                                  typename material_container::view_type,
-                                  typename surface_container::view_type,
-                                  typename volume_finder::view_type>;
-
-    using const_view_type =
+    using view_t = dmulti_view<dvector_view<volume_type>,
```

Comment:
This is a bit tricky, maybe I just haven't figured out the correct setup yet, but the problem with that is that it would lead to a ambiguous call to detray::get_data() from the one which uses the "viewable" SFINAE and the manually defined one for the `detector` type

Reply:
Ok, then let's leave it like it is for the moment